specification/compute/resource-manager/Microsoft.Compute/stable/2021-07-01/communityGallery.json

```diff
@@ -0,0 +1,558 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "CommunityGalleryServiceClient",
+    "description": "Community Gallery Service Client.",
+    "version": "2021-07-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/locations/{location}/communityGalleries/{publicGalleryName}": {
+      "get": {
```

Comment:
>  "get": { [](http://example.com/codeflow?start=4&length=10)

I see no PUT. How is a resource created?

Reply:
PUT is from gallery publisher, and it is in the 2021-07-01/gallery.json. I have not added communityGallery publishing field yet because it requires a new api version. We want to add proxy resource first, to unblock arm manifest development so we can test the consumption side first. Then we can rollout the PUT change in next api version 2021-10-01

Reply:
Do you mean PUT will be done via Microsoft.Compute/galleries resource type? It cannot be done that way since it's a different resource type.

Why not use other methods (GET, PATCH, DELETE) from Microsoft.Compute type if you are adding a field for communityGallery there?

Reply:
Yeah, PUT will be done via Microsoft.Compute/galleries. For gallery owner, they can GET with resource type Microsoft.Compute/galleries. For the audience who is shared with the gallery, they can not do GET on original gallery for security issue and we bypass RBAC. So we created fake proxy resource 'community galleries' for them to do GET. That's the intention.

For other methods (GET, PATCH, DELETE) of Microsoft.Compute/galleries, they are not ARM-faced. For the people who are shared with community gallery, we hope they can just call ARM and get the resource community gallery that is shared to them. That's another reason we want to do fake proxy resource 'community galleries'.

Reply:
I would suggest add POST on a gallery resource and make it unauthorized method. Will that work?

Reply:
You mean, this is for gallery owner or gallery sharee? so you mean, the POST is for public to fetch community gallery and it is ARM-faced? I think that's not what my team designed.

Reply:
Yes, I meant a POST for gallery sharee. I think I will need to understand your scenarios/requirements better.

What do you mean GET, PATCH, DELETE of Microsoft.Compute/galleries are not ARM-faced? Are they not going though ARM?

Reply:
So the case is that my team has two services, CAPS and CrpPir. Caps gallery call is ARM-faced. User can create a Caps gallery. And Caps replicate gallery to regional service CrpPir. And partner team consume the gallery from CrpPir. Those consumption calls are internal. Now we want to use CrpPir gallery as community gallery and share the gallery to everyone. And we need to expose such API to ARM and bypass linked access check. So this community gallery APIs are created for such purpose.

Reply:
What linked access check needs to be bypassed? What I see is a bunch for Microsoft.Compute/galleries/images/versions/write, which does not impact GET.

What I am suggesting is
1. PUT Microsoft.Comute/galleries to create a community gallery
2. Create a resource type like Microsoft.Compute/galleries/getCommunityGallery routing to CrpPir
3. POST Microsoft.Compute/galleries/{communityGalleryName]/getCommunityGallery to return the community gallery

Then you don't need to put ID in the response and POST, and it can have its own RBAC or no RBAC. Will this work?

Reply:
Hey Jin, can I set up a meeting to talk about this change? I think it would be more efficient for us to communicate.

Reply:
Yes. Please add armapireview as optional. Thanks.